features/admin.connections.v1/components/edit/settings/custom-authenticator-settings.tsx

```diff
@@ -0,0 +1,529 @@
+                        })
+                    );
+
+                    return;
+                }
+
+                dispatch(
+                    addAlert({
+                        description: t(
+                            "authenticationProvider:notifications.getIDP.genericError.description"
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t("authenticationProvider:notifications.getIDP.genericError.message")
+                    })
+                );
+            });
+    };
+
+    /**
+    * This function is used to get the custom federated authenticator details which includes endpoint
+    * configurations that need to be accessed from the "Settings" tab.
+    *
+    * @param customFederatedAuthenticatorId - Custom federated authenticator id.
+    */
+    const getCustomFederatedAuthenticator = (customFederatedAuthenticatorId: string) => {
+        getFederatedAuthenticatorDetails(connector?.id, customFederatedAuthenticatorId)
+            .then((data: FederatedAuthenticatorListItemInterface) => {
+                const endpointAuth: EndpointConfigFormPropertyInterface = {
+                    authenticationType: data?.endpoint?.authentication?.type,
+                    endpointUri: data?.endpoint?.uri
+                };
+
+                setAuthenticatorEndpoint(endpointAuth);
+            })
+            .catch((error: IdentityAppsApiException) => {
+                if (error.response && error.response.data && error.response.data.description) {
+                    dispatch(
+                        addAlert({
+                            description: t("authenticationProvider:notifications.getIDP.error.description", {
+                                description: error.response.data.description
+                            }),
+                            level: AlertLevels.ERROR,
+                            message: t("authenticationProvider:notifications.getIDP.error.message")
+                        })
+                    );
+
+                    return;
+                }
+
+                dispatch(
+                    addAlert({
+                        description: t(
+                            "authenticationProvider:notifications.getIDP.genericError.description"
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t("authenticationProvider:notifications.getIDP.genericError.message")
+                    })
+                );
+            });
+    };
+
+    const validateForm = (values: EndpointConfigFormPropertyInterface): any => {
+        const errors: Partial<EndpointConfigFormPropertyInterface> = {};
+
+        if (!values?.endpointUri) {
+            errors.endpointUri = t("actions:fields.endpoint.validations.empty");
+        }
+        if (URLUtils.isURLValid(values?.endpointUri)) {
+            if (!URLUtils.isHttpsUrl(values?.endpointUri)) {
```

Comment:
Using this method caused several issues when the user input is a deep link or a string with a combination of dots and some other characters. We keep this method only for compatibility purposes.

@DonOmalVindula did we introduce a new method to be used instead of this one? afair, we did, but I couldn't find it from the code base.